### PR TITLE
[14.0] Add sale_product_seasonality

### PR DESCRIPTION
**Pull request by simahawk**
- [x] Depends on https://github.com/OCA/product-attribute/pull/855

### CODE REVIEW

File: sale_product_seasonality/i18n/fr.po [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: sale_product_seasonality
+#: model:ir.model.fields,field_description:sale_product_seasonality.field_sale_order__date_end
+msgid "Date End"
+msgstr ""
```
Comment (jcoux) [inline sale_product_seasonality/i18n/fr.po]:
```suggestion
msgstr "Date de fin"
```

File: sale_product_seasonality/i18n/fr.po [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#. module: sale_product_seasonality
+#: model:ir.model.fields,field_description:sale_product_seasonality.field_sale_order____last_update
+#: model:ir.model.fields,field_description:sale_product_seasonality.field_sale_order_line____last_update
+msgid "Last Modified on"
+msgstr "Dernière modification le"
+
+#. module: sale_product_seasonality
+#: model:ir.ui.menu,name:sale_product_seasonality.sale_seasonal_config
+msgid "Product seasonal config"
+msgstr ""
```
Comment (jcoux) [inline sale_product_seasonality/i18n/fr.po]:
```suggestion
msgid "Product seasonal config"
msgstr "Configuration des produits de saison"
```

File: sale_product_seasonality/i18n/fr.po [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+#. module: sale_product_seasonality
+#: model:ir.ui.menu,name:sale_product_seasonality.sale_seasonal_config_line
+msgid "Product seasonal config lines"
+msgstr ""
```
Comment (jcoux) [inline sale_product_seasonality/i18n/fr.po]:
```suggestion
msgstr "Lignes de configuration des produits de saison"
```

File: sale_product_seasonality/i18n/fr.po [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+#. module: sale_product_seasonality
+#: model:ir.model,name:sale_product_seasonality.model_sale_order_line
+msgid "Sales Order Line"
+msgstr "Ligne de bons de commande"
+
+#. module: sale_product_seasonality
+#: model:ir.model.fields,field_description:sale_product_seasonality.field_sale_order__season_allowed_product_ids
+#: model:ir.model.fields,field_description:sale_product_seasonality.field_sale_order_line__season_allowed_product_ids
+msgid "Season allowed products"
+msgstr ""
```
Comment (jcoux) [inline sale_product_seasonality/i18n/fr.po]:
```suggestion
msgstr "Produits de saison autorisés"
```

File: sale_product_seasonality/i18n/fr.po [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+#. module: sale_product_seasonality
+#: model_terms:ir.ui.view,arch_db:sale_product_seasonality.view_order_form
+msgid "Seasonal allowed Products"
+msgstr ""
```
Comment (jcoux) [inline sale_product_seasonality/i18n/fr.po]:
```suggestion
msgstr "Produits de saison autorisés"
```

File: sale_product_seasonality/i18n/fr.po [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+#. module: sale_product_seasonality
+#: model:ir.model.fields,field_description:sale_product_seasonality.field_sale_order__seasonal_config_id
+msgid "Seasonal configuration"
+msgstr ""
```
Comment (jcoux) [inline sale_product_seasonality/i18n/fr.po]:
```suggestion
msgid "Product seasonal configuration"
msgstr "Configuration des produits de saison"
```

File: sale_product_seasonality/models/sale_order.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+from odoo import api, fields, models
+
+from ..utils import roundTime
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    # TODO: any better name for this field?
+    date_end = fields.Datetime()
```
Comment (jcoux) [inline sale_product_seasonality/models/sale_order.py]: Maybe `commitment_date_end` as we use `commitment_date` as start date?
```suggestion
    commitment_date_end = fields.Datetime()
```

File: sale_product_seasonality/models/sale_order.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    seasonal_config_id = fields.Many2one(
+        string="Seasonal configuration",
```
Comment (jcoux) [inline sale_product_seasonality/models/sale_order.py]:
```suggestion
        string="Product seasonal configuration",
```

File: sale_product_seasonality/models/sale_order.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        for sale in self:
+            value = [(5, 0)]
+            if sale.commitment_date:
+                products = sale._get_allowed_products(sale.commitment_date)
+                value = [(6, 0, products.ids)]
+            sale.season_allowed_product_ids = value
+
+    def _round_dates(self):
+        """TODO"""
+        commitment_date = roundTime(dt=self.commitment_date, minutes=5)
```
Comment (jcoux) [inline sale_product_seasonality/models/sale_order.py]: Maybe we want to have the number of minutes in settings to be able to round for 5,10,30 or whatever?

File: sale_product_seasonality/models/sale_order.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 Camptocamp SA
+# @author: Julien Coux <julien.coux@camptocamp.com>
```
Comment (jcoux) [inline sale_product_seasonality/models/sale_order.py:3]:
```suggestion
# @author: Damien Crier <damien.crier@camptocamp.com>
# @author: Julien Coux <julien.coux@camptocamp.com>
```

File: sale_product_seasonality/readme/CONTRIBUTORS.rst [hunk 11]
```diff
@@ -0,0 +1,2 @@
+* Simone Orsi <simone.orsi@camptocamp.com>
```
Comment (jcoux) [inline sale_product_seasonality/readme/CONTRIBUTORS.rst:2]:
```suggestion
# Damien Crier <damien.crier@camptocamp.com>
* Simone Orsi <simone.orsi@camptocamp.com>
```

**Review by jcoux — CHANGES_REQUESTED** [review]

File: sale_product_seasonality/models/sale_order.py (hunk 9 above)
Comment (simahawk) [reply sale_product_seasonality/models/sale_order.py]: good point, I add a TODO for the next round, not doing  it now

**Review by jcoux — APPROVED** [review]
Technically good for me: thanks for changes :+1:

File: sale_product_seasonality/readme/CONFIGURE.rst [hunk 12]
```diff
@@ -0,0 +1 @@
+TODO
```
Comment (damdam-s) [inline sale_product_seasonality/readme/CONFIGURE.rst]: still to do?

**Review by damdam-s — APPROVED** [review]

**Pull request merged**